src/app/services/project_service.js

```diff
@@ -716,6 +719,53 @@ angular
         return databases;
     }
 
+
+    function buildGroupTree(groups, nodes, select) {
+        var groups = {}
```

Comment:
It looks like you're shadowing the variable `groups` here, so the `groups` defined from arguments will never be used as it is overwritten. Do you need the `groups` that comes in from args?

Reply:
I don't need it.  I originally thought I did.  I've removed it.